python/rmm/_lib/memory_resource.pyx

```diff
@@ -167,101 +163,93 @@ cdef class FixedSizeMemoryResource(MemoryResource):
         pass
 
 
-cdef class FixedMultiSizeMemoryResource(MemoryResource):
+cdef class BinningMemoryResource(MemoryResource):
     def __cinit__(
         self,
-        MemoryResource upstream,
-        size_t size_base=2,
-        size_t min_size_exponent=18,
-        size_t max_size_exponent=22,
-        size_t initial_blocks_per_size=128
+        MemoryResource upstream_mr,
+        int min_size_exponent=-1,
+        int max_size_exponent=-1,
```

Comment:
I will look at changing them to int in the C++. THere's not reason to have exponents larger than an int here -- in fact they could be 8-bit ints. 2^255 is a very large number.

Reply:
Change to ints.

Reply:
@harrism can we change this to `int32_t` or `int64_t` instead of just `int`? IIRC `int` is architecture / platform dependent which is problematic if we support things like ARM and POWER.

Reply:
Done.